hack/testing/test-020-olm-upgrade.sh

```diff
@@ -0,0 +1,141 @@
+#!/bin/bash
+# Given an OLM manifest, verify a green field deployment
+# of cluster logging by asserting CLO creates the resources
+# that begets the operands that make up logging.
+
+set -e
+if [ "${DEBUG:-}" = "true" ]; then
+	set -x
+fi
+
+source "$(dirname "${BASH_SOURCE[0]}" )/../lib/init.sh"
+source "$(dirname $0)/assertions"
+source "$(dirname $0)/utils"
+
+os::test::junit::declare_suite_start "${BASH_SOURCE[0]}"
+
+ARTIFACT_DIR=${ARTIFACT_DIR:-"$(pwd)/_output"}
+if [ ! -d $ARTIFACT_DIR ] ; then
+  mkdir -p $ARTIFACT_DIR
+fi
+export NAMESPACE="openshift-logging"
+VERSION=${VERSION:-4.2}
+
+cleanup(){
+  local return_code="$?"
+  set +e
+  oc logs -n ${NAMESPACE} deployment/cluster-logging-operator > $ARTIFACT_DIR/cluster-logging-operator.log 2>&1 ||:
+  
+  for item in "crd/elasticsearches.logging.openshift.io" "crd/clusterloggings.logging.openshift.io" "ns/openshift-logging" "ns/openshift-operators-redhat"; do
+    oc delete $item --wait=false --ignore-not-found
+  done
+  os::cleanup::all "${return_code}"
+  
+  exit ${return_code}
+}
+trap cleanup exit
+
+if [ -n "${IMAGE_CLUSTER_LOGGING_OPERATOR:-}" ] ; then
+  source "$(dirname $0)/../common"
+fi
+if [ -n "${IMAGE_FORMAT:-}" ] ; then
+  IMAGE_CLUSTER_LOGGING_OPERATOR=$(sed -e "s,\${component},cluster-logging-operator," <(echo $IMAGE_FORMAT))
+else
+  IMAGE_CLUSTER_LOGGING_OPERATOR=${IMAGE_CLUSTER_LOGGING_OPERATOR:-quay.io/openshift/origin-cluster-logging-operator:latest}
+fi
+
+KUBECONFIG=${KUBECONFIG:-$HOME/.kube/config}
+
+oc create -n ${NAMESPACE} -f \
+https://raw.githubusercontent.com/coreos/prometheus-operator/master/example/prometheus-operator-crd/prometheusrule.crd.yaml || :
+oc create -n ${NAMESPACE} -f \
+https://raw.githubusercontent.com/coreos/prometheus-operator/master/example/prometheus-operator-crd/servicemonitor.crd.yaml || :
+
+repo_dir="$(dirname $0)/../.."
+
+# deploy EO
+cat<<EOL | oc create -f -
+apiVersion: v1
+kind: List
+items:
+- apiVersion: v1
+  kind: Namespace
+  metadata:
+    name: openshift-operators-redhat
```

Comment:
Are we still using `openshift-operators-redhat` for 4.2?  I suggest using a variable like `EO_NS` instead of having `openshift-operators-redhat` hardcoded everywhere.

Reply:
Changes comming